### PR TITLE
tests: migrate to jest

### PR DESCRIPTION
**Pull request by anshumanv**
**What kind of change does this PR introduce?**
tests

**Did you add tests for your changes?**
yes

**If relevant, did you update the documentation?**
NA

**Summary**
- tests migrated to jest
- created simple jest config

**Does this PR introduce a breaking change?**
no

**Other information**

### CODE REVIEW

File: jest.config.js [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+
+  // Indicates whether each individual test should be reported during the run
+  // verbose: undefined,
+
+  // An array of regexp patterns that are matched against all source file paths before re-running tests in watch mode
+  // watchPathIgnorePatterns: [],
+
+  // Whether to use watchman for file crawling
+  // watchman: true,
+};
```
Comment (alexander-akait) [inline jest.config.js:194]: Need need extra code, just keep only necessary options

**Review by alexander-akait — COMMENTED** [review]
One note

File: jest.config.js (hunk 1 above)
Comment (anshumanv) [reply jest.config.js:5]: fixed

File: package.json [hunk 2]
```diff
@@ -13,7 +13,7 @@
     "changelog": "npm run changelog:generate && npm run changelog:add",
     "changelog:add": "git add CHANGELOG.md",
     "changelog:generate": "github_changelog_generator --future-release $npm_package_version",
-    "coverage": "nyc --no-clean mocha",
+    "coverage": "nyc --no-clean jest --coverage",
```
Comment (alexander-akait) [inline package.json:16]: We don't need nyc here, we don't have extra process (i.e. `child_process`/`execa`/etc), so let's remove it

**Review by snitin315 — APPROVED** [review]

**Pull request merged**